test/parallel/test-assert-deep.js

```diff
@@ -1220,3 +1220,34 @@ assert.throws(
 
   assertNotDeepOrStrict(a, b);
 }
+
+// check URL
+{
+  const a = new URL('http://foo');
+  const b = new URL('http://bar');
+
+  assertNotDeepOrStrict(a, b);
```

Comment:
I wonder if we want a test for the error message, due to this being blocked for exactly that reason.

Reply:
Added a test for error message, pls have a look.